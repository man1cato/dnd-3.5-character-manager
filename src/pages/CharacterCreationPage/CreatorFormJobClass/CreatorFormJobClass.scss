@import '../CharacterCreationPage.scss';


.CreatorFormJobClass__group {
   @extend .CreatorForm__group;
   > :first-child {
      width: 35%;
      padding-right: $s-size;
   }
   > :last-child {
      width: 65%;
   }
   button {
      max-width: 80px;
   }
}

.CreatorFormJobClass__group--top {
   @extend .CreatorFormJobClass__group;
   align-items: flex-start;
} 