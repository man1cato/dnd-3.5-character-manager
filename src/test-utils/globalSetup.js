import fs from 'fs'
import getApi from '../utils/getApi'

export default async () => {
   const api = await getApi()
   
   fs.writeFile('src/test-utils/fixtures/api.json', JSON.stringify(api), (err) => {
      if (err) throw err
   })
}