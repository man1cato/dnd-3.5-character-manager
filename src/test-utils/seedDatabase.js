import _ from 'lodash'
import profile from './fixtures/profile'
import database from '../firebase/firebase'


export const characterOne = {
   ..._.omit(profile, ['id'])
}

export const characterTwo = {
   name: 'Korg',
   gender: 'Male',
   race: 'recoDD3gAaQXU3A56',
   jobClass: 'recbxccm8pp3fTmQw',
   alignment: 'Lawful Evil',
   level: 5,
   iconUrl: 'images/orc.svg'
}

export const characterThree = {
   name: 'Rylia',
   gender: 'Female',
   race: 'recFEpQ9zuD9Yqu7l',
   jobClass: 'recQrfCKtksnnSsaZ',
   alignment: 'Chaotic Neutral',
   level: 3,
   iconUrl: 'images/halfling.svg'
}

export const userOne = {
   uid: 'userOne',
   characterIds: ['characterOne', 'characterTwo']
}

export const userTwo = {
   uid: 'userTwo',
   characterIds: ['characterThree']
}


export default async () => {
   await database.ref('users').remove()
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[0]}`).set(characterOne)
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[1]}`).set(characterTwo)
   await database.ref(`users/${userTwo.uid}/profiles/${userTwo.characterIds[0]}`).set(characterThree)
}