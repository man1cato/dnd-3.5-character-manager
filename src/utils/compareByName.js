export default (a,b) => {
    const aName = a.name.toUpperCase();
    const bName = b.name.toUpperCase();
    if (aName < bName) { return -1 };
    if (aName > bName) { return 1 };
    return 0;
}