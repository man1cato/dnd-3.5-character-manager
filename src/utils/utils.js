import _ from 'lodash'


export const convertTextToArray = (string) => _.compact(string.split(/\n/))

export const findItemById = (storeItemArray, itemId) => storeItemArray.find((item) => item.id === itemId)

export const apiObjectToArray = (obj) => _.orderBy(Object.keys(obj).map((id) => ({
	id,
	...obj[id]
})), ['name'], ['asc'])