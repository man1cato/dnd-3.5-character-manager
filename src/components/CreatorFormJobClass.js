import React, { useState, useEffect } from 'react'
import _ from 'lodash'
import { Field, ErrorMessage } from 'formik'
import { schools, alignments } from '../utils/staticData'
import { apiObjectToArray } from '../utils/utils'


const validateProhibitedSchools = (selectedSchool) => (value) => {
	let error;
	if (selectedSchool === 'Divination') {
		if (value.length !== 1) {
			error = 'You must select one school whose magic use will be prohibited';
		}
	} else {
		if (selectedSchool !== 'Universal' && value.length !== 2) {
			error = 'You must select two schools whose magic use will be prohibited';
		}
	}
	return error;
}

const CreatorFormJobClass = ({
	values, 
	races,
	jobClasses, 
	handleChange, 
	handleMultiSelect, 
	setFieldValue, 
	setFieldError,
	validateForm
}) => {
	const [selectedJobClass, setSelectedJobClass] = useState(jobClasses[values.jobClass])

	useEffect(() => {
		if (selectedJobClass.name === 'Paladin') {
			setFieldValue('alignment', 'Lawful Good')
		} else if (selectedJobClass.name === 'Wizard') {
			setFieldValue('school', 'Universal')
		} else {
			setFieldValue('school', null)
			setFieldValue('prohibitedSchools', [])
		}
	}, [selectedJobClass])

	return (
	<>
		<div className="form-group--35">
			<h4>Job Class:</h4>
			<Field
				className="select"
				name="jobClass"
				component="select"
				onChange={(e) => {
					setSelectedJobClass(jobClasses[e.target.value])
					handleChange(e)
				}}
			>
				{apiObjectToArray(jobClasses).map((jobClass, i) => (
					<option
						key={`jobClass${i}`}
						value={jobClass.id}
					>
						{jobClass.name}
					</option>
				))}
			</Field>
		</div>

		<div className="form-group--35">
			<h4>Alignment:</h4>
			{selectedJobClass.name === "Paladin" ?
				<div>Lawful Good</div>
				:
				<Field 
					className="select" 
					name="alignment" 
					component="select" 
				>
					{alignments.map((alignment, i) => (
						<option 
							value={alignment} 
							key={`alignment${i}`}
						>
							{alignment}
						</option>
					))}
				</Field>
			}
		</div>		

		<div className="form-group--35 align-top">
			<h4>Hit Die:</h4>
			<div>{selectedJobClass.hitDie}</div>
		</div>

		<div className="form-group--35 align-top">
			<h4>Proficiencies:</h4>
			<div>{selectedJobClass.proficiencies.join(", ")}</div>
		</div>

		<div className="form-group--35 align-top">
			<h4>Bonus Languages:</h4>
			<Field 
				className="multi-select" 
				name="bonusLanguages" 
				component="select" 
				multiple 
				onChange={(e) => {handleMultiSelect(e, setFieldValue)}}
			>
				{_.orderBy(races[values.race].bonusLanguages).map((language) => (
					<option 
						value={language} 
						key={language}
					>
						{language}
					</option>
				))}
			</Field>                                    
		</div>

		<div className="form-group--35">
			<h4>Deity:</h4>
			<Field 
				className="text-input" 
				name="deity" 
				placeholder="(Optional) Enter the name of a Deity" 
			/>                                  
		</div>
		
		{selectedJobClass.name === 'Wizard' && (
			<div className="form-group--35">
				<h4>School:</h4>
				<Field 
					className="select" 
					name="school" 
					component="select" 
					onChange={(e) => {
						if (e.target.value === 'Universal') { 
							setFieldError('prohibitedSchools', undefined)	
							setFieldValue('prohibitedSchools', [])
						}						
						handleChange(e)
					}}
				>
					{schools.map((school, i) => (
						<option 
							value={school} 
							key={`school${i}`}
						>
							{school}
						</option>
					))}
				</Field>
			</div>
		)}

		{selectedJobClass.name === 'Wizard' && values.school !== 'Universal' && (
			<>
				<div className="form-group--35 align-top">
					<h4>Prohibited School(s):</h4>
					<Field 
						className="multi-select"
						name="prohibitedSchools" 
						component="select" 
						multiple 
						validate={validateProhibitedSchools(values.school)}
						onChange={(e) => {handleMultiSelect(e, setFieldValue)}}
					>
						{_.difference(schools, [values.school, 'Universal', 'Divination']).map((school, i) => (
							<option 
								value={school} 
								key={`prohibited${i}`}
							>
								{school}
							</option>
						))}
					</Field>
				</div>
				<ErrorMessage className="form-group--error" name="prohibitedSchools" component="div"/>
			</>
		)}
	</>
)
}

export default CreatorFormJobClass
