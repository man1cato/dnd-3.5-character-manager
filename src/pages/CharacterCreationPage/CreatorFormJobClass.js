import React, { useEffect } from 'react'
import _ from 'lodash'
import { Field, ErrorMessage } from 'formik'

import { schools, alignments } from '../../utils/staticData'
import { apiObjectToArray, calcStartingGold } from '../../utils/utils'


const validateProhibitedSchools = (selectedSchool) => (value) => {
	let error
	if (selectedSchool === 'Divination') {
		if (value.length < 1) {
			error = 'You must select one school whose magic use will be prohibited'
		}
		if (value.length > 1) {
			error = 'Select only one school whose magic use will be prohibited'
		}
	} else {
		if (selectedSchool !== 'Universal' && value.length < 2) {
			error = 'You must select two schools whose magic use will be prohibited'
		}
		if (selectedSchool !== 'Universal' && value.length > 2) {
			error = 'Select only two schools whose magic use will be prohibited'
		}
	}
	return error
}

const CreatorFormJobClass = ({
	values, 
	selectedRace,
	selectedJobClass, 
	setSelectedJobClass,
	jobClasses, 
	handleChange, 
	handleMultiSelect, 
	setFieldValue, 
	setFieldError
}) => {
	const { school } = values

	useEffect(() => {
		if (selectedJobClass.name === 'Paladin') {
			setFieldValue('alignment', 'Lawful Good')
		} else if (selectedJobClass.name === 'Wizard') {
			setFieldValue('school', 'Universal')
		} else {
			setFieldValue('school', null)
			setFieldValue('prohibitedSchools', [])
		}
	}, [selectedJobClass])

	return (
		<div className="container--body">			
			<div className="form-group--35">
				<h4>Job Class:</h4>
				<Field
					className="select"
					name="jobClass"
					component="select"
					onChange={(e) => {
						setSelectedJobClass(jobClasses[e.target.value])
						setFieldValue('startingGold', 0)
						handleChange(e)
					}}
				>
					{apiObjectToArray(jobClasses).map((jobClass, i) => (
						<option
							key={`jobClass${i}`}
							value={jobClass.id}
						>
							{jobClass.name}
						</option>
					))}
				</Field>
			</div>

			<div className="form-group--35">
				<h4>Alignment:</h4>
				{selectedJobClass.name === "Paladin" ?
					<div>Lawful Good</div>
					:
					<Field 
						className="select" 
						name="alignment" 
						component="select" 
					>
						{alignments.map((alignment, i) => (
							<option 
								value={alignment} 
								key={`alignment${i}`}
							>
								{alignment}
							</option>
						))}
					</Field>
				}
			</div>		

			<div className="form-group--35 align-top">
				<h4>Hit Die:</h4>
				<div>{selectedJobClass.hitDie}</div>
			</div>

			<div className="form-group--35">
				<h4>Starting Gold:</h4>
				<div className="input-group">
					<input
						className="number-input"
						type="number"
						value={values.startingGold}
						onChange={(e) => setFieldValue('startingGold', e.target.value)}
					/>
					<div> gp</div>
					<button
						className="button"
						type="button"
						onClick={() => setFieldValue('startingGold', calcStartingGold(selectedJobClass.name))}
					>
						Roll
					</button>
				</div>
			</div>

			<div className="form-group--35 align-top">
				<h4>Proficiencies:</h4>
				<div>{selectedJobClass.proficiencies.join(", ")}</div>
			</div>

			<div className="form-group--35 align-top">
				<h4>Bonus Languages:</h4>
				<Field 
					className="multi-select" 
					name="bonusLanguages" 
					component="select" 
					multiple 
					onChange={(e) => {handleMultiSelect(e, setFieldValue)}}
				>
					{_.orderBy(selectedRace.bonusLanguages).map((language) => (
						<option 
							value={language} 
							key={language}
						>
							{language}
						</option>
					))}
				</Field>                                    
			</div>

			<div className="form-group--35">
				<h4>Deity:</h4>
				<Field 
					className="text-input" 
					name="deity" 
					placeholder="(Optional) Enter the name of a Deity" 
				/>                                  
			</div>
		
			{selectedJobClass.name === 'Wizard' && (
				<div className="form-group--35">
					<h4>School:</h4>
					<Field 
						className="select" 
						name="school" 
						component="select" 
						onChange={(e) => {
							if (e.target.value === 'Universal') { 
								setFieldError('prohibitedSchools', undefined)	
								setFieldValue('prohibitedSchools', [])
							}						
							handleChange(e)
						}}
					>
						{schools.map((school, i) => (
							<option 
								value={school} 
								key={`school${i}`}
							>
								{school}
							</option>
						))}
					</Field>
				</div>
			)}

			{selectedJobClass.name === 'Wizard' && school !== 'Universal' && (
				<>
					<div className="form-group--35 align-top">
						<h4>Prohibited School(s):</h4>
						<Field 
							className="multi-select"
							name="prohibitedSchools" 
							component="select" 
							multiple 
							validate={validateProhibitedSchools(school)}
							onChange={(e) => {handleMultiSelect(e, setFieldValue)}}
						>
							{_.difference(schools, [school, 'Universal', 'Divination']).map((school, i) => (
								<option 
									value={school} 
									key={`prohibited${i}`}
								>
									{school}
								</option>
							))}
						</Field>
					</div>
					<ErrorMessage className="form-group--error" name="prohibitedSchools" component="div"/>
				</>
			)}
		</div>
	)
}

export default CreatorFormJobClass
