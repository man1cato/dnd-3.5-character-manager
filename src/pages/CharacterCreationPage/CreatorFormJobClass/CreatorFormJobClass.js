import React, { useEffect } from 'react'
import _ from 'lodash'
import { Field, ErrorMessage } from 'formik'

import { schools, alignments } from '../../../utils/staticData'
import { apiObjectToArray, calcStartingGold } from '../../../utils/utils'
import './CreatorFormJobClass.scss'


const validateProhibitedSchools = selectedSchool => value => {
	let error
	if (selectedSchool === 'Divination') {
		if (value.length < 1) {
			error = 'You must select one school whose magic use will be prohibited'
		}
		if (value.length > 1) {
			error = 'Select only one school whose magic use will be prohibited'
		}
	} else {
		if (selectedSchool !== 'Universal' && value.length < 2) {
			error = 'You must select two schools whose magic use will be prohibited'
		}
		if (selectedSchool !== 'Universal' && value.length > 2) {
			error = 'Select only two schools whose magic use will be prohibited'
		}
	}
	return error
}

const CreatorFormJobClass = ({
	values, 
	jobClasses, 
	handleChange, 
	handleMultiSelect, 
	setFieldValue, 
	setFieldError
}) => {
	const { race, jobClass, school } = values

	useEffect(() => {
		if (jobClass.name === 'Paladin') {
			setFieldValue('alignment', 'Lawful Good')
		} else if (jobClass.name === 'Wizard') {
			setFieldValue('school', 'Universal')
		} else {
			setFieldValue('school', null)
			setFieldValue('prohibitedSchools', [])
		}
	}, [jobClass])

	return (
		<div className="container--body">			
			<div className="CreatorFormJobClass__group">
				<h4>Job Class:</h4>
				<select
					className="select"
					defaultValue={_.findKey(jobClasses, jobClass)}
					onChange={(e) => {
						setFieldValue('jobClass', jobClasses[e.target.value])
						setFieldValue('startingGold', 0)
					}}
				>
					{_.map(apiObjectToArray(jobClasses), jobClass => (
						<option
							key={jobClass.id}
							value={jobClass.id}
						>
							{jobClass.name}
						</option>
					))}
				</select>
			</div>

			<div className="CreatorFormJobClass__group">
				<h4>Alignment:</h4>
				{jobClass.name === "Paladin" ?
					<div>Lawful Good</div>
					:
					<Field 
						className="select" 
						name="alignment" 
						component="select" 
					>
						{alignments.map((alignment, i) => (
							<option 
								value={alignment} 
								key={`alignment${i}`}
							>
								{alignment}
							</option>
						))}
					</Field>
				}
			</div>		

			<div className="CreatorFormJobClass__group--top">
				<h4>Hit Die:</h4>
				<div>{jobClass.hitDie}</div>
			</div>

			<div className="CreatorFormJobClass__group">
				<h4>Starting Gold:</h4>
				<div className="CreatorForm__input-group">
					<input
						className="number-input"
						type="number"
						value={values.startingGold}
						onChange={(e) => setFieldValue('startingGold', Number(e.target.value))}
					/>
					<div> gp</div>
					<button
						className="button"
						type="button"
						onClick={() => setFieldValue('startingGold', calcStartingGold(jobClass.name))}
					>
						Roll
					</button>
				</div>
			</div>

			<div className="CreatorFormJobClass__group--top">
				<h4>Proficiencies:</h4>
				<div>{jobClass.proficiencies.join(", ")}</div>
			</div>

			<div className="CreatorFormJobClass__group--top">
				<h4>Bonus Languages:</h4>
				<Field 
					className="multi-select" 
					name="bonusLanguages" 
					component="select" 
					multiple 
					onChange={(e) => handleMultiSelect(e, setFieldValue)}
				>
					{_.sortBy(race.bonusLanguages).map(language => (
						<option 
							value={language} 
							key={language}
						>
							{language}
						</option>
					))}
				</Field>                                    
			</div>

			<div className="CreatorFormJobClass__group">
				<h4>Deity:</h4>
				<Field 
					className="text-input" 
					name="deity" 
					placeholder="(Optional) Enter the name of a Deity" 
				/>                                  
			</div>
		
			{jobClass.name === 'Wizard' && (
				<div className="CreatorFormJobClass__group">
					<h4>School:</h4>
					<Field 
						className="select" 
						name="school" 
						component="select" 
						onChange={(e) => {
							if (e.target.value === 'Universal') { 
								setFieldError('prohibitedSchools', undefined)	
								setFieldValue('prohibitedSchools', [])
							}						
							handleChange(e)
						}}
					>
						{schools.map((school, i) => (
							<option 
								value={school} 
								key={`school${i}`}
							>
								{school}
							</option>
						))}
					</Field>
				</div>
			)}

			{jobClass.name === 'Wizard' && school !== 'Universal' && (
				<>
					<div className="CreatorFormJobClass__group--top">
						<h4>Prohibited School(s):</h4>
						<Field 
							className="multi-select"
							name="prohibitedSchools" 
							component="select" 
							multiple 
							validate={validateProhibitedSchools(school)}
							onChange={(e) => {handleMultiSelect(e, setFieldValue)}}
						>
							{_.difference(schools, [school, 'Universal', 'Divination']).map((school, i) => (
								<option 
									value={school} 
									key={`prohibited${i}`}
								>
									{school}
								</option>
							))}
						</Field>
					</div>
					<ErrorMessage className="CreatorForm__error" name="prohibitedSchools" component="div"/>
				</>
			)}
		</div>
	)
}

export default CreatorFormJobClass
