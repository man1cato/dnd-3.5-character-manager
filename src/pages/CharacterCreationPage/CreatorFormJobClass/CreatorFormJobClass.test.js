import React from 'react'
import _ from 'lodash'

import CreatorFormJobClass from './CreatorFormJobClass'
import { apiData, renderWithRedux, FormikWrapper } from '../../../test-utils/utils'
import { characterOne } from '../../../test-utils/seedDatabase'


const handleChange = jest.fn()
const handleMultiSelect = jest.fn()
const setFieldValue = jest.fn()
const setFieldError = jest.fn()

let props

beforeAll(async () => {
   const api = await apiData()
   
   props = {
      values: { 
         race: api.races[characterOne.race],
         jobClass: api.jobClasses[characterOne.jobClass],
         school: characterOne.school 
      },
      jobClasses: api.jobClasses,
      handleChange,
      handleMultiSelect,
      setFieldValue,
      setFieldError
   }
})


test('should render CreatorFormJobClass', () => {
   const { container } = renderWithRedux(<CreatorFormJobClass {...props} />, null, FormikWrapper)
   expect(container.firstChild).toMatchSnapshot()
})

