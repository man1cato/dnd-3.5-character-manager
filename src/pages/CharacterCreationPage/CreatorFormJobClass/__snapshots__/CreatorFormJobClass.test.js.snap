// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreatorFormJobClass 1`] = `
<form>
  <div
    class="container--body"
  >
    <div
      class="CreatorFormJobClass__group"
    >
      <h4>
        Job Class:
      </h4>
      <select
        class="select"
      >
        <option
          value="recbxccm8pp3fTmQw"
        >
          Barbarian
        </option>
        <option
          value="rec0GDUzwsKLYEyqM"
        >
          Bard
        </option>
        <option
          value="recawEdTMnIzdNnwR"
        >
          Cleric
        </option>
        <option
          value="recz6owTBNig43RWq"
        >
          Druid
        </option>
        <option
          value="reckjdfZ5f86z4Rtz"
        >
          Fighter
        </option>
        <option
          value="recVqqb9HsZRyEAC8"
        >
          Monk
        </option>
        <option
          value="recjjJTiERQ1ocuNh"
        >
          Paladin
        </option>
        <option
          value="recKnUPMQEmoCGcag"
        >
          Ranger
        </option>
        <option
          value="recQrfCKtksnnSsaZ"
        >
          Rogue
        </option>
        <option
          value="recniRmHZzspPDAFn"
        >
          Sorcerer
        </option>
        <option
          selected=""
          value="recBCWC99ohXXKqwL"
        >
          Wizard
        </option>
      </select>
    </div>
    <div
      class="CreatorFormJobClass__group"
    >
      <h4>
        Alignment:
      </h4>
      <select
        class="select"
        name="alignment"
      >
        <option
          value="Lawful Good"
        >
          Lawful Good
        </option>
        <option
          value="Neutral Good"
        >
          Neutral Good
        </option>
        <option
          value="Chaotic Good"
        >
          Chaotic Good
        </option>
        <option
          value="Lawful Neutral"
        >
          Lawful Neutral
        </option>
        <option
          value="True Neutral"
        >
          True Neutral
        </option>
        <option
          value="Chaotic Neutral"
        >
          Chaotic Neutral
        </option>
        <option
          value="Lawful Evil"
        >
          Lawful Evil
        </option>
        <option
          value="Neutral Evil"
        >
          Neutral Evil
        </option>
        <option
          value="Chaotic Evil"
        >
          Chaotic Evil
        </option>
      </select>
    </div>
    <div
      class="CreatorFormJobClass__group--top"
    >
      <h4>
        Hit Die:
      </h4>
      <div>
        D4
      </div>
    </div>
    <div
      class="CreatorFormJobClass__group"
    >
      <h4>
        Starting Gold:
      </h4>
      <div
        class="CreatorForm__input-group"
      >
        <input
          class="number-input"
          type="number"
          value=""
        />
        <div>
           gp
        </div>
        <button
          class="button"
          type="button"
        >
          Roll
        </button>
      </div>
    </div>
    <div
      class="CreatorFormJobClass__group--top"
    >
      <h4>
        Proficiencies:
      </h4>
      <div>
        Club, Dagger, Light Crossbow, Heavy Crossbow, Quarterstaff
      </div>
    </div>
    <div
      class="CreatorFormJobClass__group--top"
    >
      <h4>
        Bonus Languages:
      </h4>
      <select
        class="multi-select"
        multiple=""
        name="bonusLanguages"
      >
        <option
          value="Draconic"
        >
          Draconic
        </option>
        <option
          value="Gnoll"
        >
          Gnoll
        </option>
        <option
          value="Gnome"
        >
          Gnome
        </option>
        <option
          value="Goblin"
        >
          Goblin
        </option>
        <option
          value="Orc"
        >
          Orc
        </option>
        <option
          value="Sylvan"
        >
          Sylvan
        </option>
      </select>
    </div>
    <div
      class="CreatorFormJobClass__group"
    >
      <h4>
        Deity:
      </h4>
      <input
        class="text-input"
        name="deity"
        placeholder="(Optional) Enter the name of a Deity"
        value=""
      />
    </div>
    <div
      class="CreatorFormJobClass__group"
    >
      <h4>
        School:
      </h4>
      <select
        class="select"
        name="school"
      >
        <option
          value="Abjuration"
        >
          Abjuration
        </option>
        <option
          value="Conjuration"
        >
          Conjuration
        </option>
        <option
          value="Divination"
        >
          Divination
        </option>
        <option
          value="Enchantment"
        >
          Enchantment
        </option>
        <option
          value="Evocation"
        >
          Evocation
        </option>
        <option
          value="Illusion"
        >
          Illusion
        </option>
        <option
          value="Necromancy"
        >
          Necromancy
        </option>
        <option
          value="Transmutation"
        >
          Transmutation
        </option>
        <option
          value="Universal"
        >
          Universal
        </option>
      </select>
    </div>
    <div
      class="CreatorFormJobClass__group--top"
    >
      <h4>
        Prohibited School(s):
      </h4>
      <select
        class="multi-select"
        multiple=""
        name="prohibitedSchools"
      >
        <option
          value="Abjuration"
        >
          Abjuration
        </option>
        <option
          value="Conjuration"
        >
          Conjuration
        </option>
        <option
          value="Enchantment"
        >
          Enchantment
        </option>
        <option
          value="Evocation"
        >
          Evocation
        </option>
        <option
          value="Illusion"
        >
          Illusion
        </option>
        <option
          value="Necromancy"
        >
          Necromancy
        </option>
      </select>
    </div>
  </div>
</form>
`;
